src/main/kotlin/com/dansplugins/factionsystem/command/faction/addmember/MfFactionAddMemberCommand.kt

```diff
@@ -0,0 +1,110 @@
+package com.dansplugins.factionsystem.command.faction.addmember
+
+import com.dansplugins.factionsystem.MedievalFactions
+import com.dansplugins.factionsystem.faction.MfFaction
+import com.dansplugins.factionsystem.faction.MfFactionMember
+import com.dansplugins.factionsystem.player.MfPlayer
+import dev.forkhandles.result4k.onFailure
+import org.bukkit.ChatColor
+import org.bukkit.command.Command
+import org.bukkit.command.CommandExecutor
+import org.bukkit.command.CommandSender
+import org.bukkit.command.TabCompleter
+import org.bukkit.entity.Player
+import preponderous.ponder.command.dropFirst
+import java.util.logging.Level
+
+class MfFactionAddMemberCommand(private val plugin: MedievalFactions) : CommandExecutor, TabCompleter {
+
+    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<out String>): Boolean {
+        if (!sender.hasPermission("mf.force.addmember")) {
+            sender.sendMessage("${ChatColor.RED}${plugin.language["CommandFactionAddMemberNoPermission"]}")
+            return true
+        }
+        if (sender !is Player) {
+            sender.sendMessage("${ChatColor.RED}${plugin.language["CommandFactionAddMemberNotAPlayer"]}")
+            return true
+        }
+        if (args.isEmpty()) {
+            sender.sendMessage("${ChatColor.RED}${plugin.language["CommandFactionAddMemberUsage"]}")
+            return true
+        }
+
+        // get target player
+        val targetPlayer = plugin.server.getOfflinePlayer(args[0])
+        if (!targetPlayer.hasPlayedBefore()) {
+            sender.sendMessage("${ChatColor.RED}${plugin.language["CommandFactionAddMemberInvalidTargetPlayer"]}")
+            return true
+        }
+        val playerService = plugin.services.playerService
+        val targetMfPlayer = playerService.getPlayer(targetPlayer)
+            ?: playerService.save(MfPlayer(plugin, sender)).onFailure {
+                sender.sendMessage("${ChatColor.RED}${plugin.language["CommandFactionAddMemberFailedToSavePlayer"]}")
+                plugin.logger.log(Level.SEVERE, "Failed to save player: ${it.reason.message}", it.reason.cause)
+                return true
+            }
+
+        val factionService = plugin.services.factionService
+
+        // if target player is already in a faction, cancel
+        if (factionService.getFaction(targetMfPlayer.id) != null) {
+            sender.sendMessage("${ChatColor.RED}${plugin.language["CommandFactionAddMemberTargetPlayerAlreadyInFaction"]}")
+            return true
+            // question: should we remove the player from their current faction in this case instead of returning?
```

Comment:
Perhaps prompt to remove? This could be a question we ask the user. Maybe we need a removemember command too?

Reply:
That's a good idea. I'd rather worry about this later though. I've opened #1674 for this.

Reply:
I don't think a 'removemember' command is necessary because we can just allow the 'kick' command to be used on players in other factions when in bypass mode.